main.go

```diff
@@ -85,6 +85,13 @@ func main() {
 		log.Fatalf("Failed migrate DB %v", err)
 	}
 
+	// Update payments with preimage as an empty string to use NULL instead
+	// TODO: remove this migration in a future release and make preimage column unique
+	if err := db.Table("payments").Where("preimage = ?", "").Update("preimage", nil).Error; err != nil {
```

Comment:
And if we don't have proper migrations. Then we must release this as a breaking change with details in the release notes, otherwise we break the deployment of others.

Reply:
I vote for a proper migration setup - I don't think we can release this as a breaking change. NWC on all umbrel apps will break and it's unreasonable to ask users to manually do something to fix it.

Reply:
+1

https://gorm.io/docs/migration.html

Reply:
gorm on its own does not seem to support doing a manual migration here (note: the string column from non-nullable to nullable IS migrated automatically, but not the query to update the preimage from `''` to NULL needs to be done manually)

It looks like https://github.com/go-gormigrate/gormigrate could be the best option here, it runs inline (as opposed to atlas and golang-migrate)

Reply:
I think auto-migrate is mostly a dangerous option as you always have to be careful with the data that you have already in the DB.

and I think it's better to write explicit migrations then it is also possible to support different special SQL dialects (PG, SQLite)